### PR TITLE
4.0 ABI Fixes

### LINKED ISSUE #6123: [Bug] Change of BackgroundImage PropertyType from string to FileImageSource causes Build Errors in Rg.Plugins.Popup
### Description

In Preview Release 4.0.0 Pre 10 the BackgroundImage was changed from string to FileImageSource (and was obsoleted).

https://github.com/xamarin/Xamarin.Forms/releases/tag/beta-4.0.0-pre10.1

Checkin
https://github.com/xamarin/Xamarin.Forms/pull/4915

Now the Release Compilation fails on iOs with following message:

(_CompileToNative target) -> 
  MTOUCH : error MT2101: Can't resolve the reference 'System.String Xamarin.Forms.Page::get_BackgroundImage()', referenced from the method 'System.Void Rg.Plugins.Popup.Animations.Base.FadeBackgroundAnimation::Preparing(Xamarin.Forms.View,Rg.Plugins.Popup.Pages.PopupPage)' in 'Xamarin.Forms.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null'. 

It means it doesn't find the BackgroundImage with the string PropertyType.

### Steps to Reproduce

1.  Make new Xamarin project.
2.  Add Rg.Plugins.Popup nuget package
3.  Add following linker file (To simulate usage).

<linker>

```
    <assembly fullname="Rg.Plugins.Popup">
      <namespace fullname="Rg.Plugins.Popup.Animations.Base" />
      <type fullname="Rg.Plugins.Popup.Animations.Base.*" />
    </assembly>
```

4. Compile in Release Mode

### Expected Behavior
It does compile

### Actual Behavior
It doesn't Compile

### Basic Information

- Version with issue: 4.0.0 Pre Release 10
- Last known good version: 4.0.0 Pre Release 9
- IDE:
- Platform Target Frameworks: 
  - iOS:  12.2
- Nuget Packages: Rg.Plugins.Popup 1.1.5.188

### Reproduction Link
[App30.zip](https://github.com/xamarin/Xamarin.Forms/files/3155959/App30.zip)

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

With #4915 we are wanting to keep all the original types the same so as to limit/remove any ABI breaks. This change snuck in 

### Issues Resolved ### 
- fixes #6123

#### Type Changed: Xamarin.Forms.Button

Modified properties:

```diff
-public ImageSource Image { get; set; }
+public FileImageSource Image { get; set; }
```

#### Type Changed: Xamarin.Forms.NavigationPage

Removed methods:

```csharp
public static ImageSource GetTitleIcon (BindableObject bindable);
public static void SetTitleIcon (BindableObject bindable, ImageSource value);
```

#### Type Changed: Xamarin.Forms.Page

Modified properties:

```diff
-public FileImageSource BackgroundImage { get; set; }
+public string BackgroundImage { get; set; }
```

#### Type Changed: Xamarin.Forms.Platform.Android.AppCompat.TabbedPageRenderer

Added methods:

```csharp

[Obsolete ("GetIconDrawable is obsolete as of 4.0.0. Please override SetTabIconImageSource instead.")]
protected virtual Android.Graphics.Drawables.Drawable GetIconDrawable (Xamarin.Forms.FileImageSource icon);

[Obsolete ("SetTabIcon is obsolete as of 4.0.0. Please use SetTabIconImageSource instead.")]
protected virtual void SetTabIcon (Android.Support.Design.Widget.TabLayout.Tab tab, Xamarin.Forms.FileImageSource icon);
protected virtual void SetTabIconImageSource (Android.Support.Design.Widget.TabLayout.Tab tab, Android.Graphics.Drawables.Drawable icon);
```

#### Type Changed: Xamarin.Forms.Platform.iOS.ColorExtensions

Added method:

```csharp
public static UIKit.UIColor FromPatternImageFromBundle (string bgImage);
```

#### Type Changed: Xamarin.Forms.Platform.MacOS.TabbedPageRenderer

Removed method:

```csharp
protected virtual AppKit.NSImage GetTabViewItemIcon (AppKit.NSImage image);
```

Added methods:

```csharp
protected virtual AppKit.NSImage GetTabViewItemIcon (string imageName);
protected virtual AppKit.NSImage GetTabViewItemIconImageSource (AppKit.NSImage image);
```

#### Type Changed: Xamarin.Forms.Platform.UWP.FormsSlider

Obsoleted fields:

```diff
 [Obsolete ("ThumbImageProperty is obsolete as of 4.0.0. Please use ThumbImageSourceProperty instead.")]

 public static Windows.UI.Xaml.DependencyProperty ThumbImageProperty;
```

Added field:

```csharp
public static Windows.UI.Xaml.DependencyProperty ThumbImageSourceProperty;
```

Obsoleted properties:

```diff
 [Obsolete ("ThumbImage is obsolete as of 4.0.0. Please use ThumbImageSource instead.")]

 public Windows.UI.Xaml.Media.Imaging.BitmapImage ThumbImage { get; set; }
```

Modified properties:

```diff
-public Windows.UI.Xaml.Media.ImageSource ThumbImage { get; set; }
+public Windows.UI.Xaml.Media.Imaging.BitmapImage ThumbImage { get; set; }
```

Added property:

```csharp
public Windows.UI.Xaml.Media.ImageSource ThumbImageSource { get; set; }
  ```

### Platforms Affected ### 

- Core/XAML (all platforms)
  

### Testing Procedure ###
There's a UI test that vets everything loads but another thing to check would be a quick look back at the PR for #4915 just to make sure nothing else was missed

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by dansiegel** [comment]
The Page Background property should not be changed back to a string and in fact really should be a generic ImageSource for instance if I want an embedded resource or some other remote resource for my page background. I get that this is causing an issue with Rg.Plugin.Popups but the the answer is for them to update.

**Comment by samhouts** [comment]
@dansiegel Rather than breaking existing properties, we chose to create new properties that are all ImageSource. So you'll still get the benefit by opting into using the new properties, and we won't cause crashes for apps that have yet to update. We hope this is a good scenario for everyone!

File: Xamarin.Forms.Platform.UAP/TabbedPageStyle.xaml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 			<Setter.Value>
 				<DataTemplate>
 					<StackPanel Orientation="Vertical" Name="TabbedPageHeaderStackPanel">
-						<Image Name="TabbedPageHeaderImage" DataContext="{Binding Icon, Converter={StaticResource ImageConverter}}" Source="{Binding Value}" Width="16" Height="16" Visibility="Collapsed" Margin="0,12,0,0"/>
+                        <Image Name="TabbedPageHeaderImage" DataContext="{Binding IconImageSource, Converter={StaticResource ImageConverter}}" Source="{Binding Value}" Width="16" Height="16" Visibility="Collapsed" Margin="0,12,0,0"/>
```
Comment (PureWeen) [inline Xamarin.Forms.Platform.UAP/TabbedPageStyle.xaml]: > ``
```suggestion
						<Image Name="TabbedPageHeaderImage" DataContext="{Binding IconImageSource, Converter={StaticResource ImageConverter}}" Source="{Binding Value}" Width="16" Height="16" Visibility="Collapsed" Margin="0,12,0,0"/>
```

**Comment by toomasz** [comment]
@PureWeen I tried to use ImageSource and it seems to be missing on XF 4.0.1.383393-nightly

[ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Xamarin.Forms.Xaml.XamlParseException: Position 79:22. Property ImageSource is not found or does not have an accessible getter
05-11 13:26:32.850 E/mono-rt (28573):   at Xamarin.Forms.Xaml.ApplyPropertiesVisitor.GetPropertyValue (System.Object xamlElement, Xamarin.Forms.Xaml.XmlName propertyName, Xamarin.Forms.Xaml.HydrationContext context, System.Xml.IXmlLineInfo lineInfo, System.Object& targetProperty) [0x000b2] in D:\a\1\s\Xamarin.Forms.Xaml\ApplyPropertiesVisitor.cs:390

**Pull request merged**